### DISCUSSION TITLE
How to disable the 'click' event for some condition

### DISCUSSION THREAD

**Discussion by ManimMadhav** [katspaugh/wavesurfer.js#3622] [Q&A]
Hi all, 
Is it possible to disable the ```click``` event for a region? I require this for a smoother workflow of my application.<br>
**Problem description:**
I have a pop-up menu that gets displayed when a user _clicks_ on a region, after creating it by using the ```enableDragSelection()``` method. However, if he clicks on a previously created region (while the pop-up menu is open for the current region), the previous region gets selected instead. This is why I want to disable the ```click``` event for previous regions in such cases. Basically keeping only the _current_ region active for clicks. Any suggestions?

**My approach:**
What I have tried doing to solve this issue is using a [flag variable](https://stackoverflow.com/questions/17402125/what-is-a-flag-variable).
```
if (flag==true){
    wsRegions.on('region-clicked', (region) => {
        //region-clicked workflow
    });
}
```

However, the flag variable does not seem to be able to stop the ```region-clicked``` event from getting triggered. WHat fixes are possible in such a case?

**Comment by katspaugh** [comment]
You can check your flag inside the click listener.

```js
wsRegions.on('region-clicked', (region) => {
  if (flag) {
    doStuff()
  } else {
    dontDoStuff()
  }
});
```
